server/service/queries.go

```diff
@@ -271,7 +272,11 @@ func (svc *Service) ModifyQuery(ctx context.Context, id uint, p fleet.QueryPaylo
 		query.Description = *p.Description
 	}
 	if p.Query != nil {
+		// TODO: delete the query results
 		query.Query = *p.Query
+		if err := svc.ds.DeleteAllResultsForQuery(ctx, &id); err != nil {
+			return nil, err
+		}
 	}
```

Comment:
Since this is backing a `PATCH` and the param could be null, I took that to mean that we will only ever send the query if it has changed. If that's a bad assumption can definitely add logic to check that tho!

Reply:
@rachaelshaw @zhumo thoughts?  TLDR; if a PATCH request to `/queries` includes osquery SQL, then should we delete the query results even if the SQL didn't change?

Reply:
I can just add the logic to check the diff, just to cover all the bases and keep things moving.

Reply:
@jahzielv Yeah we should check the SQL actually changed before deleting 👍

Reply:
done!